### ISSUE TITLE
Upgrade to GoogleTest 1.10

### ISSUE THREAD

**Issue by albestro** [eth-cscs/DLA-Future#101]
GoogleTest 1.10 has been released!
https://github.com/google/googletest/releases/tag/release-1.10.0

The new version fixes a problem that @rasolca addressed with exception: the filename was not set for `TestPartResult` and GTest was trying to print a `const char * = NULL`; the library now checks this condition and in case prints `<unknown file>`.

The API for typed tests has been changed, so it will require (minor) changes.

**Comment by albestro** [comment]
Closed by #102

**Comment by albestro** [comment]
The problem is still there after upgrade to 1.10.

I thoroughly checked what happens, and I found that this particular problem has been fixed after the release with commit https://github.com/google/googletest/commit/5395345ca4f0c596110188688ed990e0de5a181c

How do we want to address this? Is it ok for everyone to use this specific commit? I would not use  `master` to avoid continuously get the latest updates at every build (with potential different behaviors).
